src/main/java/com/linecorp/armeria/server/VirtualHostBuilder.java

```diff
@@ -64,212 +30,34 @@
-     * Binds the specified {@link Service} at the specified exact path.
-     */
-    public VirtualHostBuilder serviceAt(String exactPath, Service<?, ?> service) {
-        return service(PathMapping.ofExact(exactPath), service);
-    }
-
-    /**
-     * Binds the specified {@link Service} under the specified directory..
-     */
-    public VirtualHostBuilder serviceUnder(String pathPrefix, Service<?, ?> service) {
-        return service(PathMapping.ofPrefix(pathPrefix), service);
-    }
-
-    /**
-     * Binds the specified {@link Service} at the specified {@link PathMapping}.
-     */
-    public VirtualHostBuilder service(PathMapping pathMapping, Service<?, ?> service) {
-        services.add(new ServiceConfig(pathMapping, service, null));
-        return this;
-    }
-
-    /**
-     * Binds the specified {@link Service} at the specified {@link PathMapping}.
-     *
-     * @param loggerName the name of the {@linkplain ServiceRequestContext#logger() service logger};
-     *                   must be a string of valid Java identifier names concatenated by period ({@code '.'}),
-     *                   such as a package name or a fully-qualified class name
-     */
-    public VirtualHostBuilder service(PathMapping pathMapping, Service<?, ?> service, String loggerName) {
-        services.add(new ServiceConfig(pathMapping, service, loggerName));
-        return this;
-    }
-
-    /**
-     * Decorates all {@link Service}s with the specified {@code decorator}.
-     *
-     * @param decorator the {@link Function} that decorates a {@link Service}
-     * @param <T> the type of the {@link Service} being decorated
-     * @param <R> the type of the {@link Service} {@code decorator} will produce
-     */
-    public <T extends Service<T_I, T_O>, T_I extends Request, T_O extends Response,
-            R extends Service<R_I, R_O>, R_I extends Request, R_O extends Response>
-    VirtualHostBuilder decorator(Function<T, R> decorator) {
-
-        requireNonNull(decorator, "decorator");
-
-        @SuppressWarnings("unchecked")
-        final Function<Service<Request, Response>, Service<Request, Response>> castDecorator =
-                (Function<Service<Request, Response>, Service<Request, Response>>) decorator;
-
-        if (this.decorator != null) {
-            this.decorator = this.decorator.andThen(castDecorator);
-        } else {
-            this.decorator = castDecorator;
-        }
-
-        return this;
+        super(defaultHostname, hostnamePattern);
     }
 
     /**
      * Creates a new {@link VirtualHost}.
      */
     public VirtualHost build() {
-        final VirtualHost virtualHost = new VirtualHost(defaultHostname, hostnamePattern, sslContext, services);
-        return decorator != null ? virtualHost.decorate(decorator) : virtualHost;
-    }
-
-    @Override
-    public String toString() {
-        return VirtualHost.toString(getClass(), defaultHostname, hostnamePattern, sslContext, services);
+        return build();
```

Comment:
Please check `mvn test` before you push commits. And, please check CI result after you pushed commits. You can find CI links in bottom layer of this PR and the last CI link is here. https://travis-ci.org/line/armeria/builds/157646781